_posts/2022-03-17-ols-4-participant-nadine-spychala.md

```diff
@@ -0,0 +1,113 @@
+     
+- informs the user about possible use cases, caveats, and anything that is important to keep in mind when applying the measures,     
+
+- explains the measures in the first place in a tutorial-like fashion, including commonalities and differences that will be showcased in example models,  
+
+- guides the user in drawing conclusions about the measures by providing suitable visualizations of the outcomes,     
+
+- is as stable as possible, i.e., using best practices from software engineering, such as test-driven development, version control, and implementation across different computers and working environments.
+
+## Key understanding and accomplishments
+While I couldn’t tackle yet the development of educational resources of my library, I managed to finalize software that allows applying one or a set of different measures to one or multiple datasets using only one line of code – an achievement I am extremely happy about! For now, one measure of emergence is implemented and can be applied. As the current code is scalable, adding further measures will be very easy. Moreover, the output of this one line of code delivers a variable which stores both results and parameters so as to be able to easily retrieve cases of interest and visualize them.<br>
+
+During OLS-4, I also started to finally implement some code testing and think about possible “formats” of usage (e. g., shall the library be executed via a command line interface, Docker image, or do I want the user to work directly in Python?). The feedback I got from my mentors and code reviewers in this regard was particularly helpful.<br>
+
+OLS-4 has also got me started on open science project essentials like code of conduct and contributor guidance specifications.
+
+## The main goals achieved in this project
+What I’ve done:
+
+- developed software which allows applying one or more measures to one or multiple datasets in one line of code, and which delivers a pandas dataframe storing all results alongside respective parameters,
+
+```
+
+import complex_py as cp
+
+# define what data to use/generate, and what measures of emergence 
+# to apply to those data via the following five variables: 
+
+#       model_function:      contains functions to generate/load your data
+#       model_variables:     contains parameters for generating/loading your data
+#       emergence_functions: contains functions to calculate measures of emergence
+#       measure_variables:   contains parameters for calculating emergence
+#       parameters:          contains all parameters (data- and measure-related ones) 
+
+# compute measures of emergence for all parameter combination:
+emergence_df = cp.compute_emergence(model_functions, model_variables, 
+                                    emergence_functions, measure_variables, parameters)
+
+``` 
```

Comment:
```suggestion
    ```
   import complex_py as cp
   
   # define what data to use/generate, and what measures of emergence 
   # to apply to those data via the following five variables: 
   
   #       model_function:      contains functions to generate/load your data
   #       model_variables:     contains parameters for generating/loading your data
   #       emergence_functions: contains functions to calculate measures of emergence
   #       measure_variables:   contains parameters for calculating emergence
   #       parameters:          contains all parameters (data- and measure-related ones) 
   
   # compute measures of emergence for all parameter combination:
   emergence_df = cp.compute_emergence(model_functions, model_variables, 
                                       emergence_functions, measure_variables, parameters)
   
   ``` 
```